src/routes/metrics-route.ts

```diff
@@ -0,0 +1,22 @@
+import { FastifyInstance } from 'fastify';
+import { loginMetricsData } from '../services/login-metrics';
+import { userMetricsData } from '../services/user-metrics';
+
+export default async function metrics(fastify: FastifyInstance) {
+  
+  fastify.get('/metrics', async (req, resp) => {
```

Comment:
Are you now familiar with Grafana and Prometheus at this point? This endpoint should support the prometheus format. I don't believe it is currently in that format. Or am I mistaken?

I recommend you do something like:

1. Check out https://www.npmjs.com/package/fastify-metrics. Install it using `npm install --save`
2. Register metrics as a fastify plugin. Follow the documentation. Just do the default things turning routeMetrics on.
3. You should now have a new endpoint which outputs metrics in the format when you navigate to `/prometheus` (or whatever). You should not implement any custom prometheus code yourself imo - use libraries when available please.

Then, make some views in Grafana:
1. Get cht-watchdog running locally
2. Connect watchdog to your dev machine
3. Create a new dashboard on your local watchdog
4. Try to get as many metrics as you can from the metrics that are available - for example you can get `# of logins` by looking at the number `200` response codes from the `POST /authenticate` endpoint and you don't need to do anything else
6. Send a PR with the app code, the grafana json, and screenshots

Schedule time with me if you get stuck